blockchain/blockchain.go

```diff
@@ -970,57 +923,44 @@ func (b *Blockchain) StorePending(pending *Pending) error {
 			return ErrParentDoesNotMatchHead
 		}
 
-		if existingPending, err := b.pendingBlock(txn); err == nil {
+		if existingPending, err := pendingBlock(txn); err == nil {
 			if existingPending.Block.TransactionCount >= pending.Block.TransactionCount {
-				return nil // ignore the incoming pending if it has fewer transactions than the one we already have
+				// ignore the incoming pending if it has fewer transactions than the one we already have
```

Comment:
can you please elaborate in what case it is possible to have incoming block with smaller number of transactions ?